lib/setup/profile.js

```diff
@@ -66,6 +66,110 @@ var appendScript = function(name, options) {
 };
 
 /**
+ * Chooses the correct shell init script to insert avn initialization string to,
+ * based on the following criteria:
+ * 1. Append to the one that's present, if the other doesn't exist,
+ * 2. If both exist, append to the one that is sourced by the other, or
+ * 3. If both exist, but neither sources the other,
+ *    append to .bashrc if platform is 'linux',
+ *    or append to .bash_profile if platform is 'darwin' (i.e. macOS)
+ *
+ * @private
+ * @function setup.profile.~shellInitScriptChooser
+ * @return {Promise}
+ */
+var shellInitScriptChooser = function() {
+  var bashrcFile = path.join(process.env.HOME, '.bashrc');
+  var bashProfileFile = path.join(process.env.HOME, '.bash_profile');
+  var bashrcExists;
+  var bashProfileExists;
+
+  // Check if .bashrc exists
+  var bashrcPromise = new Promise(function(resolve) {
+    fs.access(bashrcFile, function(err) {
+      bashrcExists = !err;
+      resolve();
+    });
+  });
```

Comment:
Since we have `mz`, you should be able to just do:

```js
var bashrcPromise = fs.access(bashrcFile).then(function() { bashrcExists = true; });
```